esphome/components/template/text/__init__.py

```diff
@@ -24,11 +24,11 @@
 
 def validate(config):
     if CONF_LAMBDA in config:
-        if config[CONF_OPTIMISTIC]:
+        if config[CONF_OPTIMISTIC] and config[CONF_OPTIMISTIC]:
             raise cv.Invalid("optimistic cannot be used with lambda")
         if CONF_INITIAL_VALUE in config:
             raise cv.Invalid("initial_value cannot be used with lambda")
-        if CONF_RESTORE_VALUE in config:
+        if CONF_RESTORE_VALUE in config and config[CONF_RESTORE_VALUE]:
```

Comment:
Since `CONF_RESTORE_VALUE` has a default, it will *always* be present.
```suggestion
        if config[CONF_RESTORE_VALUE]:
```

Reply:
Right. Thank you.

Reply:
Is it difficult to add a test? Could you please point me a file to do that?

Reply:
There are already tests for `template`, you can just add your sample config in the file `tests/components/template/common.yaml`. Append to the existing `text:` section.

Reply:
When adding a test like this, I recommend adding the test *without* your fix, run the test (`script/test_build_components -c template`) and verify failure, then add the fix, run again and verify success.